coderd/database/dbpurge/dbpurge.go

```diff
@@ -72,13 +75,15 @@ func New(ctx context.Context, logger slog.Logger, db database.Store, clk quartz.
 	go func() {
 		defer close(closed)
 		defer ticker.Stop()
+		// Force an initial tick immediately.
+		ticker.Reset(time.Nanosecond)
```

Comment:
Won't this set the ticker's new period to a nanosecond?

I believe this is the idiomatic way to tick immediately:

```go
ticker := time.NewTicker(period)
defer ticker.Stop()
for ; true; <- ticker.C {
    ...
}
```

Reply:
> Won't this set the ticker's new period to a nanosecond?

Yes, briefly until the second prong of the `select` is reached. Then the ticker is stopped and `doTick()` executes. At the end of `doTick()`, the ticker is reset with a longer delay.

Reply:
The correction reset happens in `doTick`, so yes and no. This does however introduce a race where we may tick two times before instead of once on startup.

1. tick reset nano
2. tick
3. <-ticker.C
4. tick
5. ticker.Stop()
6. doTick()
7. <-ticker.C (immediately after because of 4)

Reply:
The "immediate tick" idiom you mention may not play nicely with `<-ctx.Done()`

Reply:
Ah yeah, that's very possible. I didn't think about it when I commented but original code is also not impervious to this race.

An alternative non-racy way is to kick it off with a different channel:

```
var tickerC <-chan time.Time
initC := make(chan time.Time, 1)
initC <- time.Time{}
close(initC)
tickerC = initC

// ...

	case tick := tickerC:
		tickerC = ticker.C
		doTick(...)
```

Reply:
An alternative is to just call `doTick()` immediately before the `for-select` loop.

Reply:
Get out of here @johnstcn with your good and reasonable ideas!

Reply:
I think the "initCh" approach definitely has merits, but not sure if it's necessary here as the context  is still available in `doTick`. :)